timesketch/api/v1/resources.py

```diff
@@ -0,0 +1,463 @@
+
+        Returns:
+            Response in json format (instance of flask.wrappers.Response)
+        """
+        if not meta:
+            meta = dict()
+        if not model_fields:
+            try:
+                model_fields = self.fields_registry[model.__tablename__]
+            except AttributeError:
+                model_fields = self.fields_registry[model[0].__tablename__]
+
+        schema = {
+            'meta': meta,
+            'objects': [marshal(model, model_fields)]
+        }
+        response = jsonify(schema)
+        response.status_code = status_code
+        return response
+
+
+class SketchListResource(ResourceMixin, Resource):
+    """Resource for listing sketches."""
+    def __init__(self):
+        self.parser = reqparse.RequestParser()
+        self.parser.add_argument('name', type=str, required=True)
+        self.parser.add_argument('description', type=str, required=False)
+
+    @login_required
+    def get(self):
+        """Handles GET request to the resource.
+
+        Returns:
+            List of sketches (instance of flask.wrappers.Response)
+        """
+        # TODO: Handle offset parameter
+        sketches = Sketch.all_with_acl()
+        paginated_result = sketches.paginate(1, 10, False)
+        meta = {
+            'next': paginated_result.next_num,
+            'previous': paginated_result.prev_num,
+            'offset': paginated_result.page,
+            'limit': paginated_result.per_page
+        }
+        if not paginated_result.has_prev:
+            meta['previous'] = None
+        if not paginated_result.has_next:
+            meta['next'] = None
+        result = self.to_json(paginated_result.items, meta=meta)
+        return result
+
+
+class SketchResource(ResourceMixin, Resource):
+    """Resource to get a sketch."""
+    @login_required
+    def get(self, sketch_id):
+        """Handles GET request to the resource.
+
+        Returns:
+            A sketch in JSON (instance of flask.wrappers.Response)
+        """
+        sketch = Sketch.query.get_with_acl(sketch_id)
+        return self.to_json(sketch)
+
+
+class ViewListResource(ResourceMixin, Resource):
+    """Resource to create a View."""
+    @login_required
+    def post(self, sketch_id):
+        """Handles POST request to the resource.
+
+        Args:
+            sketch_id: Integer primary key for a sketch database model
+
+        Returns:
+            A view in JSON (instance of flask.wrappers.Response)
+        """
+        form = build_form(request, SaveViewForm)
+        if form.validate_on_submit():
+            sketch = Sketch.query.get_with_acl(sketch_id)
+            view = View(
+                name=form.name.data, sketch=sketch, user=current_user,
+                query_string=form.query.data,
+                query_filter=json.dumps(form.filter.data))
+            db_session.add(view)
+            db_session.commit()
+            return self.to_json(view, status_code=201)
```

Comment:
Maybe use semantic class definition for the status codes e.g.
HTTP_STATUS_CODE_CREATED 201